src/main/java/org/checkstyle/autofix/recipe/UpperEllRecipe.java

```diff
@@ -43,23 +64,112 @@ public String getDescription() {
+
+        private final List<CheckstyleViolation> violations;
+        private String currentFileName;
+
+        UpperEllVisitor(List<CheckstyleViolation> violations) {
+            this.violations = violations;
+        }
+
+        @Override
+        public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext ctx) {
+            this.currentFileName = cu.getSourcePath().toString();
+            return super.visitCompilationUnit(cu, ctx);
+        }
+
         @Override
         public J.Literal visitLiteral(J.Literal literal, ExecutionContext ctx) {
             J.Literal result = super.visitLiteral(literal, ctx);
             final String valueSource = result.getValueSource();
 
             if (valueSource != null && valueSource.endsWith("l")
-                    && result.getType() == JavaType.Primitive.Long) {
+                    && result.getType() == JavaType.Primitive.Long
+                    && isAtViolationLocation(result)) {
+
                 final String numericPart = valueSource.substring(0, valueSource.length() - 1);
-                final String newValueSource = numericPart + "L";
-                result = result.withValueSource(newValueSource);
+                result = result.withValueSource(numericPart + "L");
+            }
+
+            return result;
+        }
+
+        private boolean isAtViolationLocation(J.Literal literal) {
+            final J.CompilationUnit cursor = Objects.requireNonNull(getCursor()
+                    .firstEnclosing(J.CompilationUnit.class));
+
+            final int line = computeLinePosition(cursor, literal, getCursor());
+            final int column = computeColumnPosition(cursor, literal, getCursor());
+
+            return violations.stream().anyMatch(violation -> {
+                return violation.getLine() == line
+                        && violation.getColumn() == column
+                        && Path.of(violation.getFileName()).equals(Path.of(currentFileName));
+            });
+        }
+
+        private int computePosition(
+                J tree,
+                J targetElement,
+                Cursor cursor,
+                Function<String, Integer> positionCalculator
+        ) {
+            final TreeVisitor<?, PrintOutputCapture<TreeVisitor<?, ?>>> printer =
+                    tree.printer(cursor);
+
+            final PrintOutputCapture<TreeVisitor<?, ?>> capture =
+                    new PrintOutputCapture<>(printer) {
+                        @Override
+                        public PrintOutputCapture<TreeVisitor<?, ?>> append(String text) {
+                            if (targetElement.isScope(getContext().getCursor().getValue())) {
+                                targetElement.getPrefix();
+                                super.append(targetElement.getPrefix().getWhitespace());
+                                throw new CancellationException();
```

Comment:
why do we need this exception? it is from concurrent package, which is not related to the current logic